### ISSUE TITLE
Lerna imports fail in Jest

### ISSUE THREAD

**Issue by davidgljay** [lerna/lerna#380]
Lerna's package symlinking works find in msot contexts, but creates the following error in Jest:

```
 SyntaxError: Unexpected token import

      at transformAndBuildScript (node_modules/jest-runtime/build/transform.js:284:10)
      at Object.<anonymous> (ReplyButton.js:2:56)
      at Object.<anonymous> (Replies.js:2:46)
```

This error is only produced for import statements pointing to Lerna symlinks, not to normal package imports. Here is the file being tested:

```
import React from 'react'
import ReplyButton from './ReplyButton'
import CommentContent from 'coral-plugin-commentcontent'
import {name} from './package.json'

const Replies = (props) => <div
className={name + '-container'}
style={props.styles && props.styles.container}>
  <ReplyButton {...props}/>
  {
    props.replies && props.replies.map((reply) => {
      return <CommentContent key={reply.item_id} content = {reply.content} />
    })
  }
</div>

export default Replies
```

The test:

```
import React from 'react'
import {shallow, mount} from 'enzyme'
import {expect} from 'chai'
import Replies from '../Replies'

describe('Replies', () => {
  it('should render a reply', () => {
    const render = shallow(<Replies
        postItem={(post) => console.log(post)}
        updateItem={(e) => console.log(e.target.value)}
        item_id={'1'}
        replies={[{content: 'reply text', item_id: '2'}]}/>)
    expect(render.contains('reply text')).to.be.truthy
  })
})
```

and the local package.json:

```
{
  "name": "coral-plugin-replies",
  "version": "0.0.1",
  "description": "Enables replies on coral comments.",
  "main": "Replies.js",
  "scripts": {
    "test": "jest"
  },
  "keywords": [
    "coral"
  ],
  "author": "The Coral Project",
  "license": "MIT",
  "dependencies": {
    "babel-plugin-transform-async-to-generator": "^6.8.0",
    "babel-polyfill": "^6.13.0",
    "babel-preset-es2015": "6.13.0",
    "react": "15.3.2",
    "react-dom": "15.3.2",
    "react-redux": "^4.4.5",
    "coral-plugin-commentbox": "^0.0.1",
    "coral-plugin-commentcontent": "^0.0.1"
  },
  "devDependencies": {
    "jest": "^16.0.1"
  },
  "jest": {
    "scriptPreprocessor": "./node_modules/babel-jest"
  }
}
```

I have run lerna-bootstrap, and symlinks work in other contexts. Apologies if I'm missing something basic.

**Comment by chrislloyd** [comment]
@davidgljay have you installed `babel-jest` in your dependencies? It doesn't look like it's in your `package.json`.

**Comment by thejameskyle** [comment]
This is an issue specific to other tools (Babel & Jest). Lerna does not do any compiling work with Babel (or with anything else).

**Comment by davidgljay** [comment]
Thanks! It was a babel issue after all :)

**Comment by kujenga** [comment]
@davidgljay Do you mind sharing the solution that you found? I'm running into this issue as well.

**Comment by evocateur** [comment]
If your lerna-managed packages need a transpiler, it needs to run in `prepublish` and you probably should run `lerna bootstrap` prior to running `npm test` (such as from a `pretest` lifecycle phase).

**Comment by AllenFang** [comment]
the `modulePaths` in `jest` config will be a workaround

**Comment by bobbwhy** [comment]
I am having this issue.
I am working on a project and try to run the tests with `lerna run test`.

I get this: 
` Couldn't find preset "@babel/es2015" relative to directory "/Volumes/sites/apis/pheme/pheme_modules/app"`
(I renamed packages in the settings to [project_name]_modules for auto fill in convenience when using terminal) 

I have tried the modulesPath option, but the error is always the same, showing the same module search path.

I have all babel deps bootstrapped with the --hoist option, so they are installed in the lerna root.
The test command is in the app package as: 
` "test": "jest --runInBand",`
and the jest config in package.json is below: (You will see that I set the modulePath as root, just to see if i could get a different error);
`
"jest": {
    "transform": { 
      "^.+\\.jsx?$": "babel-jest"
    },
    "modulePaths": ["/"],
    "testRegex": "/test/index.js"
  }
`
I also tried putting a symlink from [project]/node_modules/@babel to [project]/[package]/node_modules/@babel 

Nothing.

I am now going to try and transpile ahead of testing as per suggestion above.

Any other ideas more than welcome.  I really want to make lerna work for me, but have lost many hours today on it.

thank you,
Robert

**Comment by evocateur** [comment]
Run jest from the root, not lerna. I would rename the test files to match the default pattern (*.test.js), and then it’s basically zero-conf (jest does the transpilation for you).

**Comment by bobbwhy** [comment]
@evocateur : 
Thanks again for your comment.  In the end, I went back to good ole mocha/chai for this.  
Not saying it is better, but the setup I hacked together for it months ago works here just fine.

Reason I switched was because Mocha/chai had some issues with async code a while back and Jasmine had nothing like runInBand... 

For some reason the Mocha/chai issue I had before is not a problem ( yet.. maybe was fixed?  maybe i was doing something wrong? .) and it seems to run in sequence the async stuff properly, unlike jasmine..  

best,
Robert
